transaction-service-api/src/main/java/pl/adamboguszewski/transactionservice/service/api/transaction/CreateTransactionRequest.java

```diff
@@ -0,0 +1,55 @@
+package pl.adamboguszewski.transactionservice.service.api.transaction;
+
+import lombok.Value;
+import lombok.NonNull;
+
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.UUID;
+
+@Value
+public class CreateTransactionRequest {
+
+    @NonNull
+    Long totalPrice;
+
+    @NonNull
+    List<Product> products;
+
+    @NonNull
+    Info transactionInfo;
+
```

Comment:
In Transaction class you have defined fields in order: products, totalPrice, transactionInfo.
Here your order is: totalPrice, products, transactionInfo.
Match class structures. Also we have no transactionId here, why?
Btw class structure should be better designed, I suggest: Id or other object identifier fields / private fields / 'child-related' fields (like our list of products or info).